pkg/controller/common/defaults/pod_template.go

```diff
@@ -315,3 +315,12 @@ func (b *PodTemplateBuilder) WithResources(resources corev1.ResourceRequirements
 	}
 	return b
 }
+
+func (b *PodTemplateBuilder) WithPreStopHook(handler corev1.Handler) *PodTemplateBuilder {
+	if b.Container.Lifecycle == nil {
+		b.Container.Lifecycle = &corev1.Lifecycle{}
+	}
+
+	b.Container.Lifecycle.PreStop = &handler
```

Comment:
we should make it so the user-provided lifecycle override has priority over our default